### ISSUE TITLE
LBFGS converge to NaN

### ISSUE THREAD

**Issue by lespeholt** [OpenANN/OpenANN#165]
The LBFGS optimiser sometimes converge to NaN. It seems to happen under some circumstances when the network is overfitting. I haven't been able to find a small test case. My network is like this:

Input neurons: 8000
Hidden neurons: 20 (fully connected, logistic)
Output neurons: 10 (softmax)
